src/EPR.SubsidiaryBulkUpload.Application/Services/NotificationService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Text.Json;
+using EPR.SubsidiaryBulkUpload.Application.Models;
+using EPR.SubsidiaryBulkUpload.Application.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using StackExchange.Redis;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Services;
+
+public class NotificationService(
+    ILogger<NotificationService> logger, IConnectionMultiplexer redisConnectionMultiplexer) : INotificationService
+{
+    private readonly ILogger<NotificationService> _logger = logger;
+    private readonly IDatabase _redisDatabase = redisConnectionMultiplexer.GetDatabase();
+
+    public async Task SetStatus(string key, string status)
+    {
+        await _redisDatabase.StringSetAsync(key, status);
+
+        _logger.LogInformation("Redis updated key: {key} status: {status}", key, status);
+    }
+
+    public async Task SetErrorStatus(string key, List<UploadFileErrorModel> errorsModel)
```

Comment:
Sonarqube issue: Member 'SerializeErrorsToJson' does not access instance data and can be marked as static

Reply:
Updated.